sys/parsoid.js

```diff
@@ -372,10 +372,10 @@ PSP.generateAndSave = function(hyper, req, format, currentContentRes) {
                 body: res.body[format].body
             };
             resp.headers.etag = mwUtil.makeETag(rp.revision, tid);
-            var responsePromise = self.wrapContentReq(hyper, req,
-                    P.resolve(resp), format, tid);
             return self.saveParsoidResult(hyper, req, format, tid, res)
             .then(function() {
+                var responsePromise = self.wrapContentReq(hyper, req,
+                        P.resolve(resp), format, tid);
```

Comment:
So, if I understand correctly we have to query section offsets only after we've said new offsets, right?